### PR TITLE
Initial attempt at removing iron router

### PR DESCRIPTION
**Pull request by stubailo**
**Work in progress**

There is more work to do:

1. Make sure we are completely replicating Iron Router path/query parsing
2. Test that error reporting is the same
3. Make sure we are OK with subtle API changes like no longer throwing an error when a route is accessed with the wrong method (now it just does nothing and returns the normal Meteor HTML boilerplate)

### CODE REVIEW

**Comment by stubailo** [comment]
Fixes #38 and maybe others.

**Comment by stubailo** [comment]
One more question: Should routes be case-sensitive? Right now they are.

File: test/api_tests.coffee [hunk 1]
```diff
@@ -74,28 +74,19 @@ Meteor.startup ->
...
-        test.equal result.statusCode, 404
-        test.equal response.status, 'error'
-        test.equal response.message, 'API endpoint not found'
-        next()
+      result = HTTP.get 'http://localhost:3000/api/v1/tests2/'
+      test.isTrue /__meteor_runtime_config__/.test(result.content)
 
     context 'with the default autogenerated endpoints', ->
-      Restivus.addCollection new Mongo.Collection('testautogen')
+      Restivus.addCollection new Mongo.Collection('testAutogen')
```
Comment (stubailo) [inline test/api_tests.coffee]: This is what I meant by maybe accidentally making endpoints case-sensitive. I think this conforms to standards, but might be an unwanted API change.

**Comment by kahmali** [comment]
The routes _should_ be case sensitive. I'm not sure why Iron Router chose to ignore the [RFC standard](http://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html#sec3.2.3). This could very well be API-breaking for some users, but I'm leaning towards releasing this as a major update (v0.7.0) with some other small API-breaking changes. I'm not sure if that follows semver conventions tightly enough, since technically this could be done without breaking the API, but it surely _feels_ like a major change to me. And before 1.0 I feel like semver is fairly flexible. If that's the route we choose to go, I think it will make sense to create a release branch for 0.7.0 and merge these changes there, instead of dev. Just in case any small changes need to be made on 0.6.x before this release is ready. If that seems like overkill for something where API-breaking changes could potentially be avoided, we could always conform to Iron Router's case-insensitivity for an initial release (just to get rid of all the iron router errors for users under the existing API), and release a major update after. I will gladly accept any of your advice on this.

So far, these changes are passing all my tests, including some manual tests with multiple query params, and some additional manual testing on an API of mine that uses Restivus to expose a REST API for mobile apps. The only issue I've run into so far was that I had to update from 1.0.3.2 to the latest version of Meteor in that app in order to get it to run. It says it was due to a conflict in the dependency of `webapp`, which was previously 1.1.6, but now requires 1.2.0 for `simple:json-routes`. I wasn't sure if this was really an issue worth filing on that repo. Here's the output before I updated: 



Is the dependency on the newer version of webapp necessary? I'm not even sure if the latest versions of Restivus truly support all the way back to 0.9.0. I tried testing on older releases, but I need to put together a sample app that doesn't have any other dependencies, since I get a ton of errors that prevent me from running on older releases, and I can't be sure if it's just all the other dependencies in the app. I'll try to be more specific when I have some more time to test. I'm not even sure I'm concerned with supporting anything earlier than 1.0 anyway. Is there somewhere I can see usage statistics for each version of Meteor?

**Comment by stubailo** [comment]
Interesting! I just blindly went with the default dependency on Meteor 1.1.0.2, but looks like it would be better for me to be more conservative with my dependencies. Going to fix that and release a new version!

I think your idea of releasing this as part of a bigger update is great - I don't think there is a big rush to make a backwards-compatible fix. Let me know if there is anything else I should improve in my PR.

**Comment by kahmali** [comment]
Sorry for the delay on this. I finally had a chance to finish going over everything, and I think we can start getting these changes merged in. 

All of the automated tests are passing (except for those few expected failures). As I mentioned, I also tested manually using an API from a Meteor app I'm working on, which includes endpoints that use all of the various params (URL, query, and body) and all the available HTTP methods. I extracted the API from the rest of the app into what I will likely make a Restivus sample app. I'll put that up on GitHub as soon as it's ready.  

I was able to confirm that this update fixes #35 and #43.

And finally, just to make sure I've covered all the basics, I'll address all the points you made in the initial PR. 

**1. Make sure we are completely replicating Iron Router path/query parsing**
So far so good. Manually tested with all types of params. Still need to setup some automated tests for this. Routes have been changed to be case-sensitive, which is an acceptable backwards-incompatible change, since it conforms to standards.

**2. Test that error reporting is the same**
From my manual tests, error reporting appears to work as expected. Thrown errors are returned as responses. The stack traces are much cleaner, too. Some automated tests can be added for this.

**3. Make sure we are OK with subtle API changes like no longer throwing an error when a route is accessed with the wrong method (now it just does nothing and returns the normal Meteor HTML boilerplate)**
This is something I would love to fix. Sending over a 200 response with static HTML is fairly useless to someone consuming a REST API. My goal is to provide a catchall endpoint on the base API path that users can override, in case they would like to change the default response. I still need to look into this further. This would be new functionality though, so it is outside the scope of this PR. It appears it may be related to this issue (at least the status code part of it): https://github.com/iron-meteor/iron-router/issues/1055. You actually commented on that issue a few days ago, saying it needed an issue filed on the meteor repo. I'll be glad to do that once I better understand the nature of the issue (if it isn't already done by then). I'll create a separate issue on this repo for this (and all the other 0.7.0 fixes from the [roadmap](https://trello.com/b/3FMecdcJ/restivus)) as soon as I have a chance. 

I went ahead and created a `0.7.0` branch. Would you mind resubmitting this pull request to that branch (unless you know of an easier to way for me to move it to another branch)? Once you do, I'll pull these changes in there. If you think I missed anything please just let me know.

**Pull request closed without merging**